internal_task_pollers.go

```diff
@@ -239,11 +251,21 @@ func (wtp *workflowTaskPoller) poll() (*workflowTask, error) {
 		if err != nil {
 			return nil, nil, err
 		}
-		return resp.GetHistory(), resp.GetNextPageToken(), nil
+		h := resp.GetHistory()
+		size := len(h.Events)
+		if size > 0 && atDecisionTaskCompletedEventID > 0 {
+			if h.Events[size-1].GetEventId() > atDecisionTaskCompletedEventID {
+				first := h.Events[0].GetEventId() // eventIds start from 1
+				h.Events = h.Events[:atDecisionTaskCompletedEventID-first+1]
+				if h.Events[len(h.Events)-1].GetEventType() != s.EventType_DecisionTaskCompleted {
+					return nil, nil, fmt.Errorf("newGetHistoryPageFunc: atDecisionTaskCompletedEventID(%v) "+
+						"points to event that is not DecisionTaskCompleted", atDecisionTaskCompletedEventID)
+				}
+			}
+			return h, nil, nil
```

Comment:
the nextPageToken is nil only if h.Events[size-1].GetEventId() > atDecisionTaskCompletedEventID is true.